diff --git a/template.php b/template.php
index ed9e070..53cbb38 100644
--- a/template.php
+++ b/template.php
@@ -10,6 +10,28 @@ function nexus_html_head_alter(&$head_elements) {
 }
 
 /**
+ * Implements hook_css_alter().
+ */
+function nexus_css_alter(&$css) {
+  $data = array();
+
+  if (!$cache = cache_get('nexus::excludes:css')) {
+    // Get the nexus css to exclude and set in cache:
+    $css_skip = (array) theme_get_setting('css_exclude', 'nexus');
+    foreach ($css_skip as $value) {
+      $data[$value] = $value;
+    }
+    cache_set('nexus::excludes:css', $data, 'cache', CACHE_TEMPORARY);
+  }
+  else {
+    $data = $cache->data;
+  }
+
+  // Unset our skipped CSS:
+  $css = array_diff_key($css, $data);
+}
+
+/**
  * Insert themed breadcrumb page navigation at top of the node content.
  */
 function nexus_breadcrumb($vars) {
@@ -27,7 +49,7 @@ function nexus_breadcrumb($vars) {
  * Implements template_preprocess_html().
  */
 function nexus_preprocess_html(&$vars) {
-  
+
   if (isset($vars['node'])) {
     // For full nodes.
     $vars['classes_array'][] = ($vars['node']) ? 'full-node' : '';
@@ -41,7 +63,7 @@ function nexus_preprocess_html(&$vars) {
   if (module_exists('panels') && function_exists('panels_get_current_page_display')) {
     $vars['classes_array'][] = (panels_get_current_page_display()) ? 'panels' : '';
   }
-  
+
   // Since menu is rendered in preprocess_page we need to detect it here to add body classes
   $has_main_menu = theme_get_setting('toggle_main_menu');
   $has_secondary_menu = theme_get_setting('toggle_secondary_menu');
@@ -177,7 +199,7 @@ function nexus_preprocess_page(&$vars) {
     // Make sure the shortcut link is the first item in title_suffix.
     $vars['title_suffix']['add_or_remove_shortcut']['#weight'] = -100;
   }
-  
+
   if(!theme_get_setting('nexus_feed_icons')) {
     $vars['feed_icons'] = '';
   }
diff --git a/theme-settings.php b/theme-settings.php
index 45ad00b..45620a3 100644
--- a/theme-settings.php
+++ b/theme-settings.php
@@ -111,4 +111,27 @@ function nexus_form_system_theme_settings_alter(&$form, &$form_state) {
     '#default_value' => theme_get_setting('grid_display','nexus'),
     '#description'   => t("Check this option to show Button to Show Grid columns. Uncheck to hide."),
   );
+
+  $form['nexus_settings']['css_exclude'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Exclude CSS files'),
+    '#description' => t('Enter one file per line.'),
+    // The paths are stored in an array.
+    '#default_value' => implode("\n", (array) theme_get_setting('css_exclude', 'nexus')),
+  );
+
+  // Add our custom submit function:
+  $form['#submit'][] = 'nexus_form_system_theme_settings_submit';
+}
+
+/**
+ * Form submit for nexus_form_system_theme_settings_alter().
+ * @param $form
+ *   The form.
+ * @param $form_state
+ *   The form state.
+ */
+function nexus_form_system_theme_settings_submit($form, &$form_state) {
+  $exclude = array_filter(array_map('trim', explode("\n", $form_state['values']['css_exclude'])));
+  form_set_value(array('#parents' => array('css_exclude')), $exclude, $form_state);
 }
