diff --git a/FeedsCommentProcessor.inc b/FeedsCommentProcessor.inc
index 141b89d..ab0f170 100644
--- a/FeedsCommentProcessor.inc
+++ b/FeedsCommentProcessor.inc
@@ -97,7 +97,7 @@ class FeedsCommentProcessor extends FeedsProcessor {
   /**
    * Implement expire().
    */
-  public function expire($time = NULL) {
+  public function expire(FeedsSource $source, $time = NULL) {
     if ($time === NULL) {
       $time = $this->expiryTime();
     }
