diff --git a/sites/all/modules/patched/profile2/profile2.module b/sites/all/modules/patched/profile2/profile2.module
index e384cde..df2b49e 100644
--- a/sites/all/modules/patched/profile2/profile2.module
+++ b/sites/all/modules/patched/profile2/profile2.module
@@ -531,9 +531,10 @@ function profile2_category_access($account, $type_name) {
  * @see profile2_profile2_access()
  */
 function profile2_access($op, $profile = NULL, $account = NULL) {
-  if (user_access('administer profiles', $account)) {
+  // Agrega2 admin users will never have access to all profiles at the same time.
+  /*if (user_access('administer profiles', $account)) {
     return TRUE;
-  }
+  }*/
   if ($op == 'create' || $op == 'update') {
     $op = 'edit';
   }
