diff --git a/sites/all/modules/patched/facetapi/plugins/facetapi/url_processor_standard.inc b/sites/all/modules/patched/facetapi/plugins/facetapi/url_processor_standard.inc
index a3e5366..7f29960 100644
--- a/sites/all/modules/patched/facetapi/plugins/facetapi/url_processor_standard.inc
+++ b/sites/all/modules/patched/facetapi/plugins/facetapi/url_processor_standard.inc
@@ -47,8 +47,8 @@ class FacetapiUrlProcessorStandard extends FacetapiUrlProcessor {
 
     // Appends to qstring if inactive, removes if active.
     foreach ($values as $value) {
-      if ($active && isset($active_items['"' . $value . '"'])) {
-        unset($qstring[$this->filterKey][$active_items['"' . $value . '"']['pos']]);
+      if ($active && isset($active_items[$value])) {
+        unset($qstring[$this->filterKey][$active_items[$value]['pos']]);
       }
       elseif (!$active) {
         $field_alias = rawurlencode($facet['field alias']);
@@ -63,7 +63,7 @@ class FacetapiUrlProcessorStandard extends FacetapiUrlProcessor {
         }
 
         // Adds the filter to the query string.
-        $qstring[$this->filterKey][] = $field_alias . ':"' . $value . '"';
+        $qstring[$this->filterKey][] = $field_alias . ':' . $value;
       }
     }
 
